File: pinyougou-manager-web/src/main/java/com/pinyougou/manager/controller/BrandController.java
package com.pinyougou.manager.controller;

import com.alibaba.dubbo.config.annotation.Reference;
import com.pinyougou.pojo.TbBrand;
import com.pinyougou.sellergoods.service.BrandService;
import entity.PageResult;
import entity.Result;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

/**
 * @author congzi
 * @Description: 品牌Controller
 * @create 2018-05-03
 * @Version 1.0
 */
@RestController
@RequestMapping("/brand")
public class BrandController {

    @Reference
    private BrandService brandService;

    /**
     * 查询全部品牌列表
     * @return
     */
    @RequestMapping("/findAll")
    public List<TbBrand> findAll(){
        return brandService.findAll();
    }

    /**
     * 返回分页列表
     * @param page
     * @param rows
     * @return
     */
    @RequestMapping("/findPage")
    public PageResult findAll(int page ,int rows){
        PageResult pageResult = brandService.findPage(page, rows);

        return  pageResult;
    }

    @RequestMapping("/add")
    public Result add(@RequestBody TbBrand tbBrand){

        try {
            brandService.add(tbBrand);
            return new Result(true,"添加成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false,"添加失败");
        }
    }

    /**
     * 修改品牌
     * @param tbBrand
     * @return
     */
    @RequestMapping("/update")
    public  Result update(@RequestBody TbBrand tbBrand){

        try {
            brandService.update(tbBrand);
            return new Result(true,"添加成功");

        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false,"添加失败");
        }
    }

    /**
     * 根据ID获取实体
     * @param id
     * @return
     */
    @RequestMapping("/findOne")
    public TbBrand findOne(Long id){
        return brandService.findOne(id);
    }

    /**
     * 删除操作
     * @param ids
     * @return
     */
    @RequestMapping("/delete")
    public Result delete(Long[] ids){

        try {
            brandService.delete(ids);
            return new Result(true,"添加成功");

        } catch (Exception e) {

            e.printStackTrace();
            return new Result(false,"添加失败");
        }
    }
}

File: pinyougou-manager-web/src/main/webapp/js/controller/brandController.js
/**
 * AngularJS 控制层(调用 service进行页面展示)
 */
app.controller("brandController",function ($scope,$http,$controller,brandService) {

    //继承
    $controller('baseController',{$scope:$scope});

    $scope.findAll=function(){
        brandService.findAll().success(
            function(response){
                $scope.list=response;//给列表变量赋值
            }
        );
    }


    //分页方法
    $scope.findPage=function(page,rows){
        brandService.findPage(page,rows).success(
            function(response){
                $scope.paginationConf.totalItems=response.total;//总记录数
                $scope.list=response.rows;//给列表变量赋值
            }
        );
    }


    //查询实体
    $scope.findOne=function(id){
        brandService.findOne(id).success(
            function(response){
                $scope.entity=response;
            }
        );
    }

    //增加
    $scope.save=function(){
        var object=null;
        if($scope.entity.id!=null){
            object=brandService.update($scope.entity);
        }else{
            object=brandService.add($scope.entity);
        }
        object.success(
            function(response){
                if(response.success){//如果成功
                    $scope.reloadList();//刷新列表
                }else{
                    alert(response.message);//提示错误
                }
            }
        );
    }


    //批量删除
    $scope.dele=function(){

        brandService.dele($scope.selectIds).success(
            function(response){
                if(response.success){//如果成功
                    $scope.reloadList();//刷新列表
                }else{
                    alert(response.message);//提示错误
                }
            }
        );
    }

});